### PR TITLE
Added apollo graphql with codegen

### PR DESCRIPTION
**Pull request by Shaju06**
- [x]  Setup of apollo client with folder name gql in src.
 

- [x] Moreover added graphql codegen.

- [x] Added some gql queries aswell.

- [x] Used a public grapql server endpoint.

### CODE REVIEW

**Comment by Shaju06** [comment]
@pavankpdev please review

**Comment by pavankpdev** [comment]
great, will review this today @Shaju06

**Comment by pavankpdev** [comment]
@devnm21 can you review this

**Comment by Shaju06** [comment]
@pavankpdev just resolved it

File: packages/frontend/src/gql/codegen.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { CodegenConfig } from '@graphql-codegen/cli'
+
+const config: CodegenConfig = {
+  schema: 'https://rickandmortyapi.com/graphql', // Replace it with your GraphQL server endpoint
```
Comment (pavankpdev) [inline packages/frontend/src/gql/codegen.ts:4]: Let's have this config in ENV

File: packages/frontend/src/gql/index.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { ApolloClient, InMemoryCache, createHttpLink } from '@apollo/client';
+import { onError } from '@apollo/client/link/error';
+
+const httpLink = createHttpLink({
+  uri: 'https://rickandmortyapi.com/graphql', // Replace with your GraphQL API endpoint
```
Comment (pavankpdev) [inline packages/frontend/src/gql/index.ts:5]: Let's have this config in ENV

**Review by pavankpdev — CHANGES_REQUESTED** [review]
Hey @Shaju06 a few minor changes, please have a look

**Comment by Shaju06** [comment]
@pavankpdev  Added NEXT_PUBLIC_GRAPHQL_SERVER_ENDPOINT for graphql endpoint

**Comment by pavankpdev** [comment]
Hey @Shaju06 looks like the ENV you used, wasn't added to `.example.env`, can you please do, if the `.example.env` doesn't exist please create one. 

This would help others developers who will be syncing up with your changes.

**Review by pavankpdev — APPROVED** [review]
Great, merging

**Review by pavankpdev — APPROVED** [review]
Great, merging

**Comment by pavankpdev** [comment]
@Shaju06 can you share your Discord username and if you're not in Duolearn server, please join here [https://discord.gg/nkF2SyKw](https://discord.gg/nkF2SyKw)

**Pull request merged**